lib/active_model/serializer/adapter/json/fragment_cache.rb

```diff
@@ -1,16 +1,6 @@
-require 'active_model/serializer/adapter/fragment_cache'
-module ActiveModel
-  class Serializer
-    class Adapter
-      class Json < Adapter
-        class FragmentCache
+class ActiveModel::Serializer::Adapter::Json::FragmentCache
```

Comment:
I understand that's the only way if the `inherited` hook is used.

I'm thinking that it would be better to register adapters manually, without using the `inherited` hook, i.e. have people call `ActiveModel::Serializer::Adapter.register` directly.

Perhaps some inspiration can be taken from how `ActiveJob` adapters are wired[1].

[1] https://github.com/rails/rails/blob/master/activejob/lib/active_job/queue_adapter.rb